pkg/descheduler/strategies/node_affinity.go

```diff
@@ -0,0 +1,52 @@
+package strategies
+
+import (
+	"github.com/golang/glog"
+	"github.com/kubernetes-incubator/descheduler/cmd/descheduler/app/options"
+	"github.com/kubernetes-incubator/descheduler/pkg/api"
+	"github.com/kubernetes-incubator/descheduler/pkg/descheduler/evictions"
+	nodeutil "github.com/kubernetes-incubator/descheduler/pkg/descheduler/node"
+	podutil "github.com/kubernetes-incubator/descheduler/pkg/descheduler/pod"
+	"k8s.io/api/core/v1"
+)
+
+func RemovePodsViolatingNodeAffinity(ds *options.DeschedulerServer, strategy api.DeschedulerStrategy, evictionPolicyGroupVersion string, nodes []*v1.Node) {
+
+	if !strategy.Enabled {
+		return
+	}
+
+	for _, nodeAffinity := range strategy.Params.NodeAffinityType {
+
+		glog.V(2).Infof("Executing for nodeAffinityType: %v", nodeAffinity)
+
+		switch nodeAffinity {
+		case "requiredDuringSchedulingIgnoredDuringExecution":
+
+			for _, node := range nodes {
+				glog.V(1).Infof("Processing node: %#v\n", node.Name)
+
+				pods, err := podutil.ListPodsOnANode(ds.Client, node)
```

Comment:
I think this should be ListEvictablePodsOnANode

Reply:
Yep, makes sense, fixed.